vision/spatial_transformer/spatial_transformer.jl

```diff
@@ -174,16 +174,16 @@ classifier =
         Dense(256, 10),
     ) |> dev
 
-ps = Flux.params(localization_net, classifier)
+# ps = Flux.params(localization_net, classifier)
 ## ====
 # create sampling grid
 const sampling_grid = get_sampling_grid(args[:img_size]...) |> dev
 ## ====
 
-opt = Adam(1e-4)
+opt = Flux.setup(Adam(1f-4), (localization_net, classifier))
```

Comment:
Not sure if this is too surprising. Rather than make a state tree per model, it makes one for the Tuple, and then calls `Flux.update!(opt, (localization_net, classifier), grads)` on both in one go.

Reply:
To reduce the surprise, we could destructure `grads` as well and re-tuple for the `update!` here. Slightly more verbose but more explicit.

Reply:
In the VAE I just made two optimisers etc, like so:

https://github.com/FluxML/model-zoo/pull/375/files#diff-d2fcc7489d82937b5b466b70d629d200f23003b00a0e923cfa5031eedfa54d4cR112-R113

Maybe this wants a comment saying you could also do that?

Reply:
The [PyTorch version](https://pytorch.org/tutorials/intermediate/spatial_transformer_tutorial.html) uses a single net, so if anything it seems like we should bundle them together more tightly. Maybe a custom layer?

Reply:
I'm going to call that out of scope for this PR though. Just adding explicit gradients, without changing too much. Maybe it's good enough...